### PR TITLE
Deploy instructions

### LINKED ISSUE #383: Need new guidance for VERCEL_URL and AUTH0_BASE_URL
The guidance in https://github.com/auth0/nextjs-auth0/blob/main/examples/README.md#environment-variables-1 needs to change.

We can no longer assign `VERCEL_URL` to `AUTH0_BASE_URL` using "Reference to System Environment Variable" in the Vercel dashboard.

> We previously introduced the Provided by System option as some frameworks need to map system variables like VERCEL_URL to framework prefixed variables like NEXT_PUBLIC_VERCEL_URL. You no longer need to configure this mapping because the prefixed variables are added automatically based on your Framework Preset.

Source: https://vercel.com/changelog/environments-variables-per-git-branch

We will need to suggest a way that developers can assign `VERCEL_URL` to `AUTH0_BASE_URL` when the application is running (note: we don't need `AUTH0_BASE_URL` at build time)

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### Description

Vercel has made a change to their environment variables, which requires an update to our guidance

You can no longer alias `AUTH0_BASE_URL` to `VERCEL_URL` in the Vercel dashboard.

We should now suggest users alias it in their `next.config.js` `env` property. eg

```js
// next.config.js
module.exports = {
  env: {
    AUTH0_BASE_URL: process.env.VERCEL_URL || process.env.AUTH0_BASE_URL
  }
};
```

This will rewrite instances of `process.env.AUTH0_BASE_URL` to the value of `VERCEL_URL` in the build using the DefinePlugin (and fallback to `process.env.AUTH0_BASE_URL` in development)

We also need to remove the `process.env` destructuring in https://github.com/auth0/nextjs-auth0/blob/main/src/config.ts#L410-L437 to make this work.

### References

fixes #383 
https://vercel.com/changelog/environments-variables-per-git-branch
https://nextjs.org/docs/api-reference/next.config.js/environment-variables

### Testing

Tested on kitchen sink vercel deploy

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -13,9 +13,10 @@ Go to the [Auth0 dashboard](https://manage.auth0.com/) and create a new applicat
 | Setting               | Description                                                                                                                                                            |
 | --------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | Allowed Callback URLs | Should be set to `http://localhost:3000/api/auth/callback` when testing locally or typically to `https://myapp.com/api/auth/callback` when deploying your application. |
-| Allowed Logout URLs   | Should be set to `http://localhost:3000/` when testing locally or typically to `https://myapp.com/` when deploying your application. 
+| Allowed Logout URLs   | Should be set to `http://localhost:3000/` when testing locally or typically to `https://myapp.com/` when deploying your application.                                   |
```
Comment (adamjmcgrath) [inline examples/README.md:16]: prettier added a bunch of whitespace changes

File: examples/README.md [hunk 2]
```diff
@@ -74,16 +77,30 @@ As Vercel wants one single build command, we make use of the `build:vercel` npm
 
 **Note**: Vercel runs `npm install` in the root of the repository by default, so we do not need to worry about that.
 
-
 #### Environment Variables
-Once the application is configured and deployed (the first deploy will fail because of the missing environment variables, but we need to hit deploy before continuing as we need to configure a more complex set of variables which is not possible from the import screen), move to the application's Environment Variables (Settings > Environment Variables) and ensure to configure the following variables:
-
-| Name  | Type  | Value |
```
Comment (adamjmcgrath) [inline examples/README.md:81]: You don't need to specify `Type` anymore as all variables are encrypted by default

File: examples/kitchen-sink-example/next.config.js [hunk 3]
```diff
@@ -1,4 +1,7 @@
 module.exports = {
   poweredByHeader: false,
-  target: 'serverless'
+  target: 'serverless',
+  env: {
+    AUTH0_BASE_URL: process.env.VERCEL_URL || process.env.AUTH0_BASE_URL
```
Comment (frederikprijck) [inline examples/kitchen-sink-example/next.config.js:5]: Wouldnt it make more sense to flip the order here? 

```
AUTH0_BASE_URL: process.env.AUTH0_BASE_URL || process.env.VERCEL_URL
```

In that case we set the `AUTH0_BASE_URL` to `VERCEL_URL` unless `AUTH0_BASE_URL` was explicitly set.

**Review by frederikprijck — COMMENTED** [review]
Looks good, but added one comment/question.

File: examples/kitchen-sink-example/next.config.js (hunk 3 above)
Comment (adamjmcgrath) [reply examples/kitchen-sink-example/next.config.js:5]: I don't think it makes a difference. On Vercel `AUTH0_BASE_URL` wont be set and in development `VERCEL_URL` wont be set.

I'd rather prefer the production configuration on the rare occaison where both might be set.

**Review by frederikprijck — APPROVED** [review]

**Comment by pancho111203** [comment]
@adamjmcgrath Thank you for the fix. Is this going to be released soon? I'm trying to import from master but it's not working.

**Pull request merged**